R/deprecated.R

```diff
@@ -1,7 +1,55 @@
+
+
+rlang_inform <- function(..., msg = paste0(...), frequency = "regular", id = msg) {
+  rlang::inform(
+    message = msg,
+    .frequency = frequency,
+    .frequency_id = id
+  )
+}
+rlang_warn <- function(..., msg = paste0(...), frequency = "regular", id = msg) {
+  rlang::warn(
+    message = msg,
+    .frequency = frequency,
+    .frequency_id = id
+  )
+}
+
 #' Print message for deprecated functions in Shiny
 #'
 #' To disable these messages, use `options(shiny.deprecation.messages=FALSE)`.
 #'
+#' @inheritParams lifecycle::deprecate_soft
+#' @keywords internal
+#' @describeIn soft-deprecated Passes arguments to [lifecycle::deprecate_soft()]
+shinySoftDeprecated <- function(
+  when, what, with = NULL, details = NULL, id = NULL,
+  env = rlang::caller_env(1)
+) {
+  if (is_false(getOption("shiny.deprecation.messages"))) {
+    return(invisible())
+  }
+
+  lifecycle::deprecate_soft(when, what, with = with, details = details, id = id, env = env)
+}
+#' @describeIn soft-deprecated Passes along a message directly to [rlang::warn()] with `.frequency = "regular"`
+shinySoftDeprecatedMessage <- function(
+  ...
+) {
+  if (is_false(getOption("shiny.deprecation.messages"))) {
+    return(invisible())
+  }
+
+  rlang_warn(..., frequency = "regular")
```

Comment:
I think `"always"` makes more sense here, and other places we set the `frequency`.